temporian/implementation/pandas/utils.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utility functions for the Pandas implementation."""
+from typing import Union
+from temporian.implementation.pandas.data import event
+from temporian.implementation.pandas.data import sampling
+
+
+def get_index_and_timestamp_column_names(
+    obj: Union[event.PandasEvent, sampling.PandasSampling]
+) -> tuple[list[str], str]:
```

Comment:
Does this act similarly as "Tuple" from the typing package? (same for list).

Reply:
Yep! Python docs for [typing.List](https://docs.python.org/3.10/library/typing.html#typing.List) and [typing.Tuple](https://docs.python.org/3.10/library/typing.html#typing.Tuple) declare them as deprecated since 3.9, replaced by the built-in `list` and `tuple` directives :) Same goes for `dict`, `set` and `type`. See [PEP 585](https://peps.python.org/pep-0585/).

Reply:
Python 3.7 might be a stretch, but we should be compatible with python 3.8 (https://w3techs.com/technologies/details/pl-python/3). We need to check, but maybe in this case we need  a `from __future__ import annotations`.